common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/ElasticIndexer.scala

```diff
@@ -10,12 +9,23 @@ import com.sksamuel.elastic4s.circe._
 import io.circe.Encoder
 import grizzled.slf4j.Logging
 
-class ElasticIndexer[T: Indexable](client: ElasticClient, index: Index)(
+import uk.ac.wellcome.elasticsearch.{
+  ElasticsearchIndexCreator,
+  IndexConfig,
+  NoStrictMapping
+}
+
+class ElasticIndexer[T: Indexable](client: ElasticClient,
+                                   index: Index,
+                                   config: IndexConfig = NoStrictMapping)(
```

Comment:
Do we really want a default here? I'm not sure one config is more preferable than any other.

Reply:
It would be good to have this with no default, especially as `NoStrictMapping` isn't  a common used config, and is the outlier.